.claude/hooks/test-gate.py

```
@@ -0,0 +1,424 @@
+        return 0  # "git ... commit" appeared only as prose/quoted text
+
+    for invocation in invocations:
+        if bypasses_verification(invocation["arguments"]):
+            block("Blocked (#47 gate): --no-verify / -n is not available "
+                  "to agents.",
+                  "Fix the failing tests (or the missing test update) instead.")
+
+    for invocation in invocations:
+        # Query the SAME repository the commit targets (git -C / --git-dir).
+        try:
+            hooks_path = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)